src/main/java/com/parasoft/demoapp/service/OrderService.java

```diff
@@ -61,15 +61,13 @@ public InventoryOperationRequestMessageDTO handleMessageFromResponseQueue(Invent
                 }
             }
             return null;
-        } catch (OrderNotFoundException e) {
-            // TODO will be handled in separate task(rollback the inventory)
-            log.error("Order Not Found Exception:", e);
-            return new InventoryOperationRequestMessageDTO(InventoryOperation.INCREASE,
+        } catch (OrderNotFoundException | ParameterException e) {
+            log.error("Order Exception:", e);
+            return new InventoryOperationRequestMessageDTO(InventoryOperation.NONE,
                     operationResult.getOrderNumber(),
-                    null,
                     e.getMessage());
-        } catch (ParameterException e) {
-            log.error("Parameter Exception:", e);
+        } catch (OrderStatusException e) {
+            log.error("Order Status Exception:", e);
             assert order != null;
             return new InventoryOperationRequestMessageDTO(InventoryOperation.INCREASE,
```

Comment:
When we try to modify an order status backward, should we just just use replayTo to return the error message? Why are we creating an increase message on request queue?